activestorage/lib/active_storage/analyzer/image_analyzer/image_magick.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module ActiveStorage
+  class Analyzer::ImageAnalyzer::ImageMagick < Analyzer
+    def metadata
+      read_image do |image|
+        if rotated_image?(image)
+          { width: image.height, height: image.width }
+        else
+          { width: image.width, height: image.height }
+        end
+      end
```

Comment:
It looks like this method is the same in the `ImageMagick` and `Vips` classes. Could you DRY it up?

Reply:
Will do.